app/controllers/solidus_stripe/webhooks_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require "solidus_stripe/webhook/event"
+require "stripe"
+
+module SolidusStripe
+  class WebhooksController < Spree::BaseController
+    SIGNATURE_HEADER = "HTTP_STRIPE_SIGNATURE"
+
+    skip_before_action :verify_authenticity_token, only: :create
+
+    respond_to :json
+
+    def create
+      request.body.read
+        .then { |payload| Stripe::Webhook.construct_event(payload, signature_header, secret, tolerance: tolerance) }
+        .then { |stripe_event| SolidusStripe::Webhook::Event.new(stripe_event: stripe_event) }
+        .then { |event| Spree::Bus.publish(event) }
```

Comment:
What happens when we receive an unknown event? 
Is that ignored by the bus or raises an exception?
Should we only publish registered events?

(please excuse my ignorance of the new bus on this topic)

Reply:
When an unknown event is received, an exception is raised, rendering the names of all registered events. That helps users be protected against typos. For instance, say that you subscribe by accident to an event `:"stripe.charge.succeded"` (there's a missing `e`). If we don't raise when there's an incoming `chage.succeeded` stripe event, users can have a hard time debugging.

More generally, getting an event you weren't anticipating is something unexpected, so I think an exception makes sense here.

What do you think?

Reply:
I see the point, which brings me to another question. I thought that stripe sends out all events it can, or do we have control in the dashboard over what events we want to receive?

I'd say if we have control then it's fine to raise; otherwise we risk having a bunch of exceptions from all the tens of events sent by stripe: in that case maybe we can filter the list of what we want to forward.

Reply:
My understanding is that stripe requires you to [register the events you're interested in on the dashboard](https://stripe.com/docs/webhooks/go-live#configure-webhook-settings).

Reply:
A possible pain point would be for users listening to all events:
<img src="https://user-images.githubusercontent.com/52650/215456664-56ae1f32-409c-4c08-857f-cbf8670b5875.png" width="50%" />

However, Omnes is designed with safety in mind, so it doesn't allow publishing events if they haven't been registered before. We could make some changes upstream to introduce namespace events and different strategies for checking the event name. For instance, allowing any event starting with `stripe.` if the user prefers.

I'd say we should move forward now and work on it in a later stage if we think or find out that's a valid concern. Thoughts?

Reply:
The fact that Stripe lets you list the events you wanna receive settles it for me, we just need to instruct to list only the ones supported out-of-the-box when installing (or provide the code needed to do it from the API) and that's it 🙌 

Thanks for explaining 👍